From 8fb0364bd0e19f35c20616dd1ab10aa00e08fa8f Mon Sep 17 00:00:00 2001
From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Wed, 10 Apr 2019 14:49:43 -0500
Subject: [PATCH] opkg: add target for testsuite installation

- Add target to install testsuite
- Add override of opkg binary used during test via OPKG_PATH environment
variable.

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>
---
Upstream-Status: Submitted

 Makefile.am  | 5 +++++
 tests/cfg.py | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 90cfcae..1f6a784 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,3 +18,8 @@ run-tests:
 	$(MAKE) -C tests DATADIR=@datadir@ SYSCONFDIR=@sysconfdir@ VARDIR=@localstatedir@
 
 check: run-tests
+
+install-ptest:
+	cp -r $(srcdir)/tests $(DESTDIR)
+	cp Makefile $(DESTDIR)
+	sed -e 's/^Makefile/_Makefile/' < Makefile > $(DESTDIR)/Makefile
diff --git a/tests/cfg.py b/tests/cfg.py
index 4efaff6..ca59d03 100644
--- a/tests/cfg.py
+++ b/tests/cfg.py
@@ -2,4 +2,4 @@ import os
 
 opkdir = "/tmp/opk"
 offline_root = "/tmp/opkg"
-opkgcl = os.path.realpath("../src/opkg")
+opkgcl = os.getenv('OPKG_PATH', os.path.realpath("../src/opkg"))
-- 
2.20.1

